Sage/carbon
# feat(flat-table): update keyboard navigation behaviour when rows are focusable FE-5773

### Proposed behaviour


Overrides previous keyboard navigation for focusable rows. Tabbing no longer navigates the rows, this has been replaced by up and down arrow keys.

Overrides the use-agent styling by setting `focus-visible` to `outline: none`
![image](https://github.com/Sage/carbon/assets/44157880/f11a755a-da38-4649-a914-95611d42951d)

### Current behaviour


Tab key is used to navigate focusable rows

user agent style is applied via `focus-visible` which adds a double outline to clickable rows/ cells

![image](https://github.com/Sage/carbon/assets/44157880/aac3d71c-87c2-4aba-bd13-37483ff45490)


### Checklist




#### QA


### Additional context



### Testing instructions


Stories in storybook should be enough to test (clickable/ highlightable/ selectable/ expandable), happy to build sandbox if QA request it for further testing

The following CodeSandbox is an example of the broken behaviour.
You can see the new behaviour by looking at the version in the comment by `codesandbox[bot]`.
